This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit fb88d04dcfb2eb2ba63c4f6dd924afed96d5b2f7
Author: Alex Gaynor <alex.gaynor@gmail.com>
Date:   Sat Jan 15 15:40:14 2022 -0500

    Update to heck 0.4 (#201)

Index: strum-macros/src/helpers/case_style.rs
===================================================================
--- strum-macros.orig/src/helpers/case_style.rs
+++ strum-macros/src/helpers/case_style.rs
@@ -1,4 +1,6 @@
-use heck::{CamelCase, KebabCase, MixedCase, ShoutySnakeCase, SnakeCase, TitleCase};
+use heck::{
+    ToKebabCase, ToLowerCamelCase, ToShoutySnakeCase, ToSnakeCase, ToTitleCase, ToUpperCamelCase,
+};
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub enum CaseStyle {
@@ -58,9 +60,9 @@ impl CaseStyleHelpers for syn::Ident {
         let ident_string = self.to_string();
         if let Some(case_style) = case_style {
             match case_style {
-                CaseStyle::PascalCase => ident_string.to_camel_case(),
+                CaseStyle::PascalCase => ident_string.to_upper_camel_case(),
                 CaseStyle::KebabCase => ident_string.to_kebab_case(),
-                CaseStyle::MixedCase => ident_string.to_mixed_case(),
+                CaseStyle::MixedCase => ident_string.to_lower_camel_case(),
                 CaseStyle::ShoutySnakeCase => ident_string.to_shouty_snake_case(),
                 CaseStyle::SnakeCase => ident_string.to_snake_case(),
                 CaseStyle::TitleCase => ident_string.to_title_case(),
@@ -68,7 +70,7 @@ impl CaseStyleHelpers for syn::Ident {
                 CaseStyle::LowerCase => ident_string.to_lowercase(),
                 CaseStyle::ScreamingKebabCase => ident_string.to_kebab_case().to_uppercase(),
                 CaseStyle::CamelCase => {
-                    let camel_case = ident_string.to_camel_case();
+                    let camel_case = ident_string.to_upper_camel_case();
                     let mut pascal = String::with_capacity(camel_case.len());
                     let mut it = camel_case.chars();
                     if let Some(ch) = it.next() {
Index: strum-macros/Cargo.toml
===================================================================
--- strum-macros.orig/Cargo.toml
+++ strum-macros/Cargo.toml
@@ -27,7 +27,7 @@ repository = "https://github.com/Peterna
 name = "strum_macros"
 proc-macro = true
 [dependencies.heck]
-version = "0.3"
+version = "0.4"
 
 [dependencies.proc-macro2]
 version = "1.0"
